src/Cli/dotnet/commands/dotnet-workload/update/WorkloadUpdateCommand.cs

```diff
@@ -175,6 +224,8 @@ public void UpdateWorkloads(bool includePreviews = false, DirectoryPath? offline
                         _workloadInstaller.RemoveManifestsFromInstallState(sdkFeatureBand);
                     }
 
+                    _workloadInstaller.AdjustWorkloadSetInInstallState(sdkFeatureBand, string.IsNullOrWhiteSpace(_workloadSetVersion) ? null : workloadSetVersion);
```

Comment:
That actually is not what this does. If the version isn't specified explicitly (i.e., _workloadSetVersion is null or white space), then it writes null into the install state for the workload set version, and since it doesn't serialize nulls, that means it _removes_ the version (if present) from the install state.

Reply:
I traced the code and it looked like `_workloadSetVersion` would be set here even if it wasn't specified on the command line, because there was code in one of the calling methods that read it from the advertised workload set and passed it along.

Reply:
This is the total set of writes to _workloadSetVersion:
<img width="531" alt="image" src="https://github.com/dotnet/sdk/assets/12969783/db2df5f0-7ecd-4fbf-ab60-35a3632df15b">

Two of those are clearly from the parseResult, i.e., from the command line. The third is here with this comment:
<img width="750" alt="image" src="https://github.com/dotnet/sdk/assets/12969783/b7c8a14e-dce4-4fe5-a72d-699e63e9cc0e">

In other words, if it's already pinned, we treat it as if it's pinned so that it stays in the install state file. If you run a vanilla `dotnet workload update`, it isn't set.

Reply:
I think maybe I got confused here by the difference between `_workloadSetVersion` and `workloadSetVersion`.  I commented on the same code in `WorkloadInstallCommand`, I think it would be easier to understand if this line was consistent about using `_workloadSetVersion`, and it would be good to have a comment explaining why it does so.